internal/utils/misc.go

```diff
@@ -25,19 +25,21 @@ func NamespacedName(obj client.Object) types.NamespacedName {
 // GetHashedName returns a partially hashed name for the string including up to the given length of the original name characters before the hash.
 // Input `nsName` should be formatted as `{Namespace}/{ResourceName}`.
 func GetHashedName(nsName string, length int) string {
-	hashedName := HashString(nsName)
+	hashedName := Digest(nsName)
 	// replace `/` with `-` to create a valid K8s resource name
 	resourceName := strings.ReplaceAll(nsName, "/", "-")
 	if length > 0 && len(resourceName) > length {
 		// resource name needs to be trimmed, as container port name must not contain consecutive hyphens
 		trimmedName := strings.TrimSuffix(resourceName[0:length], "-")
-		return fmt.Sprintf("%s-%s", trimmedName, hashedName[0:8])
+		return fmt.Sprintf("%s-%s", trimmedName, hashedName)
 	}
-	return fmt.Sprintf("%s-%s", resourceName, hashedName[0:8])
+	return fmt.Sprintf("%s-%s", resourceName, hashedName)
 }
 
-func HashString(str string) string {
-	h := sha256.New() // Using sha256 instead of sha1 due to Blocklisted import crypto/sha1: weak cryptographic primitive (gosec)
-	h.Write([]byte(str))
-	return strings.ToLower(fmt.Sprintf("%x", h.Sum(nil)))
+// Digest returns a 32-bit hashh of the input string.
+// The hash is represented as a capitalized hexadecimal string.
+func Digest(str string) string {
+	h := fnv.New32a()
```

Comment:
How did this not cause a change in envoy service names ?
Won't this change cause a churn in services that were deployed with v0.6 and now there will be no automated way to clean up those resources?

Reply:
Yes, that would be a problem :-(

Then we need to continue using the old way: generate a 256 hash and only use the first 8 characters of the hex string.

Reply:
Yeah, would be ideal to avoid creating this messy situation for current users w/o an easy way to clean up